invokeai/backend/model_manager/storage/migrate.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2023 The InvokeAI Development Team
+
+import shutil
+from pathlib import Path
+
+from omegaconf import OmegaConf
+
+from invokeai.app.services.config import InvokeAIAppConfig
+from invokeai.backend.util.logging import InvokeAILogger
+
+from .base import CONFIG_FILE_VERSION
+
+
+def migrate_models_store(config: InvokeAIAppConfig):
+    # avoid circular import
+    from invokeai.backend.model_manager import DuplicateModelException, InvalidModelException, ModelInstall
+    from invokeai.backend.model_manager.storage import get_config_store
+
+    app_config = InvokeAIAppConfig.get_config()
+    logger = InvokeAILogger.get_logger()
+    old_file: Path = app_config.model_conf_path
+    new_file: Path = old_file.with_name("models3_2.yaml")
+
+    old_conf = OmegaConf.load(old_file)
+    store = get_config_store(new_file)
+    installer = ModelInstall(store=store)
+    logger.info(f"Migrating old models file at {old_file} to new {CONFIG_FILE_VERSION} format")
+
+    for model_key, stanza in old_conf.items():
+        if model_key == "__metadata__":
+            assert (
+                stanza["version"] == "3.0.0"
+            ), f"This script works on version 3.0.0 yaml files, but your configuration points to a {stanza['version']} version"
+            continue
+
+        base_type, model_type, model_name = model_key.split("/")
+
+        try:
+            path = app_config.models_path / stanza["path"]
+            new_key = installer.register_path(path)
+        except DuplicateModelException:
+            # if model already installed, then we just update its info
+            models = store.search_by_name(model_name=model_name, base_model=base_type, model_type=model_type)
```

Comment:
Because of the earlier issue with `split("/")`, `model_name` could be something other than string here, but it should be `str | None`.

`base_model` is expected to be `BaseModelType` (an enum), but isn't. Same with `model_type`. These should be instantiated as members of the enum eg `base_model=BaseModelType(base_type)` or something like that.